vm/src/obj/objcomplex.rs

```diff
@@ -75,33 +72,35 @@ impl PyComplex {
         re.hypot(im)
     }
 
-    #[pymethod(name = "__add__")]
-    fn add(&self, other: PyObjectRef, vm: &VirtualMachine) -> PyResult {
+    #[inline]
+    fn op<F>(&self, other: PyObjectRef, op: F, vm: &VirtualMachine) -> PyResult
+    where
+        F: Fn(Complex64, Complex64) -> PyResult,
```

Comment:
Change this function signature into `Fn(Complex64, Complex64) -> Complex64` to reduce boiler plate code elsewhere.